### ISSUE TITLE
Request - Reorder catalog record Place and Time fields so they are in the same order all the time

### ISSUE THREAD

**Issue by falco-rk** [ArctosDB/arctos#8261]
**Describe what you're trying to do** 

If you [enter a single catalog record](https://arctos.database.museum/Bulkloader/enter.cfm), Place and Time fields are in this order: Record-Event, Event, Locality, Spatial.
If you [download a bulkload builder](https://arctos.database.museum/Bulkloader/bulkloaderBuilder.cfm) for catalog records, Place and Time fields are in this order: Locality, Spatial, Event, Record-Event.
If you [browse and edit loaded catalog records](https://arctos.database.museum/Bulkloader/browseBulk.cfm), Place and Time fields are in this order: Locality, Spatial, Event, Record-Event.
The [documentation](https://handbook.arctosdb.org/how_to/How-to-Enter-Data-for-a-Single-Record.html#place-and-time) for how to enter a single catalog record has Place and Time fields in this order: Record-Event, Event, Locality, Spatial.

This is confusing. Can we have them all in the same order? My vote for order is Record-Event, Event, Locality, Spatial but I would accept if they are in a different order as long as its consistent!

@happiah-madson @genevieve-anderegg @campmlc @dustymc

**Comment by dustymc** [comment]
I don't think there are any technical limitations involved in this, but there are a few social considerations.

1. "Place-time" data are useful for all sorts of things and I don't think there are any safe assumptions about the context in which someone might use them. Certainly catalog items are not a necessary part of the landscape.
2. There are inherit dependencies - localities cannot exist without geography, for example - and those have, up until the most recent bulkloader revision, controlled (ish, probably...) order of appearance. The "natural" order is likely the only scalable approach.
3. Some UI is user-controlled, and I fully expect and hope to have even less control of UI as Arctos grows; investments in reordering might have limited payoffs. (But certainly there are things which can easily be changed, and setting a direction - even if we can't immediately fully implement it - does seem useful.)
4. That logical order doesn't completely survive the current bulkloader/record entry UI (which is generally backwards because that's what was asked for) - eg it's possible to enter data using only event name and never acknowledge (or be forced to view) the existence of anything else.
5. downloads are probably handled differently by various things within Arctos now (I think including user's preferences) - if this is going to result in some requirement for globally ordering CSV, I'm probably also going to ask for tools and support.


>Record-Event, Event, Locality, Spatial

Both geography and localities can carry spatial data, this doesn't quite make sense to me.

Vaguely related and in anticipation of a 'thou shalt' flavored resolution, it would be super-useful to have a place for this sort of thing. (Possibly https://handbook.arctosdb.org/how_to/developer-guide.html)

**Comment by campmlc** [comment]
@mkoo  please review. This is an important UI/UX need.

**Comment by happiah-madson** [comment]
>Record-Event, Event, Locality, Spatial

>Both geography and localities can carry spatial data, this doesn't quite make sense to me.

This is the literal order of the fields in the "Create Records" data entry and the handbook. I agree with Rosie that it doesn't matter the order, _we just need it to be consistent._

**Comment by Jegelewicz** [comment]
> Both geography and localities can carry spatial data, this doesn't quite make sense to me.

Visit this locality - https://arctos.database.museum/place.cfm?action=detail&locality_id=12119001

Note that geography is at the top and there is a polygon for that
![image](https://github.com/user-attachments/assets/8507323b-5d48-4610-b2dd-57c3d0c841bf)

Personal comment - I have always hated this order becuase I get shocked by the number of records when I am only expecting one...

Which you find below:
![image](https://github.com/user-attachments/assets/333668fe-b6cb-4586-a0e1-25e62d4dcef3)

Note that the geography polygon is also included in this map, which includes the spatial data (single point of coordinates plus error) of the locality.

Does that make sense?

**Comment by happiah-madson** [comment]
> Personal comment - I have always hated this order because I get shocked by the number of records when I am only expecting one...

Hard agree!!!!! I assume I am on the wrong page and panic!!!!

**Comment by dustymc** [comment]
> geography is at the top and there is a polygon for that

See eg https://arctos.database.museum/place.cfm?action=detail&locality_id=12293239 there's geography

<img width="722" alt="Screenshot 2024-11-05 at 15 24 16" src="https://github.com/user-attachments/assets/d957872e-8bba-4294-9946-aa394595d307">

then zoom in a bit and find locality (green thing)


<img width="361" alt="Screenshot 2024-11-05 at 15 24 38" src="https://github.com/user-attachments/assets/dfed5ea4-495c-4f27-a724-5ffd5b0b2263">

(overlain by the autogenerated point-radius approximation)

"spatial data" does not define one thing. This is trying to become a change request, clarity is important. 

Here's a slightly different view of the above, note the datatypes
```

arctosprod@arctos>> \d geog_auth_rec
                                                           Table "core.geog_auth_rec"
          Column          |          Type           | Collation | Nullable |                              Default                               
--------------------------+-------------------------+-----------+----------+--------------------------------------------------------------------
 geog_auth_rec_id         | integer                 |           | not null | 
...
 spatial_footprint        | geography               |           |          | 
...



arctosprod@arctos>> \d locality
                                                                               Table "core.locality"
             Column             |            Type             | Collation | Nullable |                                           Default                                            
--------------------------------+-----------------------------+-----------+----------+----------------------------------------------------------------------------------------------
 locality_id                    | integer                     |           | not null | 
...
 locality_footprint             | geography                   |           |          | 
...

```

I'm not sure if this is important or related or not, but localities are not limited to "single point of coordinates plus error" spatial data - they are just spatial, no artificial limitations.

**Comment by mkoo** [comment]
I dont think we need to overthink this (here's my summary from the comments):
Fxn need is to have consistent order of data entry, bulkloading, and bulkediting forms. The desired order would be: Record-Event, Event, Locality, Spatial.
However given the potential constraints on the ordering, it's more important to have the order consistent across those forms. This excludes result tables since this is user controlled and other customizable interfaces.
Fair summary? 

Can at least the bulkloader and browse/edit follow the single data entry order of Record-Event, Event, Locality, Spatial ?

Place.cfm comments are not really helpful in this thread and distracting in honing the original request from Rosie so let's focus on the data entry and editing needs first. That should be a separate issue!

**Comment by happiah-madson** [comment]
>Fair summary?

From my view, yes
